### PR TITLE
Update badge addin

### PR DESCRIPTION
**Pull request by stephhazlitt**
• make a new fxn for project state badges
• update RStudio addin
• update/add deprecated versions of insert badge functions
• update NEWS
• bump version 1.0.4 -> 1.0.5

### CODE REVIEW

File: R/addins.R [hunk 1]
```diff
@@ -33,21 +33,21 @@ devex_gadget <- function() {
...
-            shiny::p("In production, but maybe in Alpha or Beta. Intended to persist and be supported.")),
+            shiny::img(src   = "https://img.shields.io/badge/Lifecycle-Dormant-%23ff7f2a"), 
+            shiny::p("The project is currently not under active development, but there are plans to redevelop.")),
           shiny::span(
-            shiny::img(src = "https://assets.bcdevexchange.org/images/badges/retired.svg"), 
-            shiny::p("No longer being used or supported, recommend not using an alternative option."))), 
-        choiceValues = c("inspiration", "exploration", "dormant", "delivery", "retired")),
+            shiny::img(src = "https://img.shields.io/badge/Lifecycle-Retired-d45500"), 
+            shiny::p("The project is no longer being used and/or supported."))), 
+        choiceValues = c("experimental", "marturing", "dormant", "stable", "retired")),
```
Comment (ateucher) [inline R/addins.R]:
```suggestion
        choiceValues = c("experimental", "maturing", "stable", "dormant", "retired")),
```

typo, and the order was different than the selection list, so choosing "stable" inserted "dormant", and vice-versa

File: R/addins.R [hunk 2]
```diff
@@ -33,21 +33,21 @@ devex_gadget <- function() {
...
+            shiny::p("The codebase is being roughed out, but finer details are likely to change.")),
           shiny::span(
-            shiny::img(src = "https://assets.bcdevexchange.org/images/badges/dormant.svg"), 
-            shiny::p("Not currently being worked on, but with plans to come back to the work in the future.")), 
+            shiny::img(src = "https://img.shields.io/badge/Lifecycle-Stable-97ca00"), 
+            shiny::p("The project is in a reliable state and major changes are unlikely to happen.")), 
           shiny::span(
-            shiny::img(src   = "https://assets.bcdevexchange.org/images/badges/delivery.svg"), 
-            shiny::p("In production, but maybe in Alpha or Beta. Intended to persist and be supported.")),
+            shiny::img(src   = "https://img.shields.io/badge/Lifecycle-Dormant-%23ff7f2a"),
```
Comment (ateucher) [inline R/addins.R]:
```suggestion
            shiny::img(src   = "https://img.shields.io/badge/Lifecycle-Dormant-ff7f2a"), 
```

The `%23` is url encoded `#`, not necessary here I don't think

File: R/devex_badge.R [hunk 3]
```diff
@@ -32,29 +32,76 @@ insert_bcgov_devex_badge <- function(project_state, cat = TRUE) {
...
 #' @param cat Use cat to print the result (\code{TRUE}; default) or return a character vector (\code{FALSE})?
 #'
 #' @return html
 #' @export
 devex_badge <- function(project_state, cat = TRUE) {
-  .Deprecated("insert_bcgov_devex_badge")
-  insert_bcgov_devex_badge(project_state = project_state, cat = cat)
+  .Deprecated("insert_bcgov_lifecycle_badge")
+  insert_bcgov_lifecycle_badge(project_state = project_state, cat = cat)
 }
```
Comment (ateucher) [inline R/devex_badge.R]: I think it's been long enough that we can entirely remove devex_badge(), and remove this commented-out code

**Review by ateucher — CHANGES_REQUESTED** [review]
This is awesome @stephhazlitt! Just a couple minor comments/tweaks that I noticed.

File: R/devex_badge.R (hunk 3 above)
Comment (stephhazlitt) [reply R/devex_badge.R]: Done, thanks for the review @ateucher!

**Comment by boshek** [comment]
We likely should increment version and add a NEWS item too.

**Comment by boshek** [comment]
Also am I missing where the descriptions of the badges are documented? I see it in the shiny app and the add-in but we likely should accommodate non-rstudio users somehow.

**Comment by stephhazlitt** [comment]
@boshek great catch, I'll add both (NEWS bullet and url to badge descriptions in docs for `insert_bcgov_lifecycle_badge()`.

**Review by ateucher — APPROVED** [review]

**Comment by ateucher** [comment]
LGTM!

**Comment by stephhazlitt** [comment]
@ateucher @boshek I think this is ready, however waiting until this [PR](https://github.com/bcgov/repomountie/pull/116) is merged so that the permanent landing url for the new badges is live.

**Pull request merged**